verilog/analysis/json_diagnostics.cc

```diff
@@ -0,0 +1,74 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "verilog/analysis/json_diagnostics.h"
+
+#include <vector>
+#include <string>
+
+#include "absl/strings/string_view.h"
+#include "common/analysis/file_analyzer.h"
+#include "common/strings/line_column_map.h"
+#include "verilog/analysis/verilog_analyzer.h"
+#include "json/value.h"
+
+namespace verilog {
+
+// Returns AnalysisPhase as JSON value.
+// Try not to change. External tools may use these values as a constant phase IDs.
+static Json::Value analysis_phase_to_json(const verible::AnalysisPhase& phase) {
+  switch (phase) {
+    case verible::AnalysisPhase::kLexPhase:
+      return "lex";
+    case verible::AnalysisPhase::kPreprocessPhase:
+      return "preprocess";
+    case verible::AnalysisPhase::kParsePhase:
+      return "parse";
+    default:
+      return "";
```

Comment:
Suggest a non-blank string like "unknown-phase".

Reply:
I've changed this to "unknown"